card/mod/01_core/set/all/references.rb

```diff
@@ -1,115 +1,103 @@
 
+PARTIAL_REF_CODE = 'P'
+
 def name_referencers link_name=nil
   link_name = link_name.nil? ? key : link_name.to_name.key
-  Card.joins( :references_to ).where card_references: { referee_key: link_name }
+  Card.joins(:references_to).where card_references: { referee_key: link_name }
 end
 
 def extended_referencers
-  # FIXME .. we really just need a number here.
+  # FIXME: .. we really just need a number here.
   (descendants + [self]).map(&:referencers).flatten.uniq
 end
 
-
 # replace references in card content
 def replace_references old_name, new_name
-  obj_content = Card::Content.new raw_content, card=self
-  obj_content.find_chunks( Card::Chunk::Reference ).select do |chunk|
-    if old_ref_name = chunk.referee_name and new_ref_name = old_ref_name.replace_part(old_name, new_name)
-      chunk.referee_name = chunk.replace_reference old_name, new_name
-      Card::Reference.where( referee_key: old_ref_name.key ).update_all referee_key: new_ref_name.key
-    end
+  obj_content = Card::Content.new raw_content, self
+  obj_content.find_chunks(Card::Chunk::Reference).select do |chunk|
+    next unless (old_ref_name = chunk.referee_name)
+    next unless (new_ref_name = old_ref_name.replace_part old_name, new_name)
+    chunk.referee_name = chunk.replace_reference old_name, new_name
+    old_references = Card::Reference.where(referee_key: old_ref_name.key)
+    old_references.update_all referee_key: new_ref_name.key
   end
 
   obj_content.to_s
 end
 
 # update entries in reference table
-def update_references rendered_content = nil, refresh = false
-  raise "update references should not be called on new cards" if id.nil?
+def update_references rendered_content=nil, refresh=false
+  raise 'update references should not be called on new cards' if id.nil?
 
   Card::Reference.delete_all_from self
 
-  # FIXME: why not like this: references_expired = nil # do we have to make sure this is saved?
-  #Card.update( id, references_expired: nil )
-  #  or just this and save it elsewhere?
-  #references_expired=nil
+  # FIXME: why not like this: references_expired = nil
+  # do we have to make sure this is saved?
+  # Card.update( id, references_expired: nil )
+  # or just this and save it elsewhere?
+  # references_expired=nil
 
-  Card.connection.execute("update cards set references_expired=NULL where id=#{id}")
-#  references_expired = nil
+  Card.connection.execute(
+    "update cards set references_expired=NULL where id=#{id}"
+  )
   expire if refresh
 
-  rendered_content ||= Card::Content.new(raw_content, card=self)
+  rendered_content ||= Card::Content.new raw_content, self
   rendered_content.find_chunks(Card::Chunk::Reference).each do |chunk|
-    if referee_name = chunk.referee_name # name is referenced (not true of commented inclusions)
-      referee_id = chunk.referee_id
```

Comment:
Chunk#referee_id was doing a full card fetch; all we need is the id.  I think this may have been slowing things down a bunch?